src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -107,37 +107,57 @@ private IEdmEntityType ComputeBaseType()
 
         private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
         {
-            if (this.entity.Key != null)
+            if (this.entity.Key == null)
             {
-                List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
-                foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
+                return null;
+            }
+
+            List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
+            foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
+            {
+                IEdmStructuralProperty structuralProperty = null;
+                if (!string.IsNullOrEmpty(keyProperty?.PropertyAlias))
+                {
+                    string[] keyPropertySegments = keyProperty.PropertyName.Split('/');
+                    if (keyPropertySegments.Length > 0)
+                    {
+                        structuralProperty = this.FindProperty(keyPropertySegments[0]) as IEdmStructuralProperty;
+                        if (structuralProperty == null)
+                        {
+                            return null;
+                        }
+
+                        structuralProperty = PropertyAliasHelpers.GetKeyProperty(structuralProperty, keyPropertySegments, keyProperty.PropertyAlias);

```

Comment:
Given that `structuralProperty` was assigned a value 5 or 6 line above here, could you add a comment above here to give someone an idea what happens when you call `PropertyAliasHelpers.GetKeyProperty` and what new value `structuralProperty` assumes